firefox-ios/Client/Frontend/Autofill/Address/Edit/EditAddressViewController.swift

```diff
@@ -35,11 +35,52 @@ class EditAddressViewController: UIViewController, WKNavigationDelegate, WKScrip
     }
 
     private func setupWebView() {
+        model.saveAction = { [weak self] completion in
+            self?.webView.evaluateJavaScript("getCurrentFormData();") { result, error in
+                if let error = error {
+                    self?.logger.log(
+                        "JavaScript execution error",
+                        level: .warning,
+                        category: .autofill,
+                        description: "JavaScript execution error: \(error.localizedDescription)"
+                    )
+                    return
+                }
+
+                guard let resultDict = result as? [String: Any] else {
+                    self?.logger.log(
+                        "Result is nil or not a dictionary",
+                        level: .warning,
+                        category: .autofill,
+                        description: "Result is nil or not a dictionary"
+                    )
+                    return
+                }
+
+                do {
+                    let jsonData = try JSONSerialization.data(withJSONObject: resultDict, options: [])
+                    let decoder = JSONDecoder()
+                    decoder.userInfo[.formatStyleKey] = FormatStyle.kebabCase
+                    let address = try decoder.decode(UpdatableAddressFields.self, from: jsonData)
+                    completion(address)
+                } catch {
+                    self?.logger.log(
+                        "Failed to decode dictionary",
+                        level: .warning,
+                        category: .autofill,
+                        description: "Failed to decode dictionary \(error.localizedDescription)"
+                    )
+                }
+            }
+        }
+
         if let url = Bundle.main.url(forResource: "AddressFormManager", withExtension: "html") {
             let request = URLRequest(url: url)
             webView.loadFileURL(url, allowingReadAccessTo: url)
             webView.load(request)
         }
+
+        webView.configuration.userContentController.add(self, name: "addressFormMessageHandler")
```

Comment:
could we put the name somewhere static please

Reply:
We can actually remove this code.